kyuubi-server/src/main/scala/org/apache/kyuubi/session/SessionLimiter.scala

```diff
@@ -95,7 +95,13 @@ class SessionLimiterImpl(userLimit: Int, ipAddressLimit: Int, userIpAddressLimit
 
   private def decrLimitCount(key: String): Unit = {
     _counters.get(key) match {
-      case count: AtomicInteger => count.decrementAndGet()
+      case count: AtomicInteger if count.get > 0 =>
+        count.synchronized {
```

Comment:
It seems like this is enough:
```
count.accumulateAndGet(1, (l, r) => if (l > 0) l - r else l)
```

Reply:
@wForget @bowenliang123 hi, could you tell me, in a user session, if the call to SessionLimiter's `increment` method fails(limited by something) or TFrontendService's `OpenSession` method fails, will it finally call the SessionLimiter's `decrement` method? If not, I think there might be some issues.

Reply:
It will always be called, even if close session is not called correctly, it will be cleaned up by `SessionManager#timeoutChecker`.

Reply:
+1

Reply:
> It seems like this is enough:
> 
> ```
> count.accumulateAndGet(1, (l, r) => if (l > 0) l - r else l)
> ```

@wForget Thanks for your answer, and I have adopted your code suggestions.